web_field_required_invisible_manager/tests/test_web_field_required_invisible_manager.py

```diff
@@ -65,9 +69,92 @@ def setUpClass(cls):
+        )
+        cls.company_city_field = cls.env["ir.model.fields"].search(
+            [("model", "=", "res.company"), ("name", "=", "city")]
+        )
+        cls.company_name_readonly = cls.env["custom.field.restriction"].create(
+            {
+                "readonly_model_id": cls.company_model_id.id,
+                "field_id": cls.company_name_field.id,
+                "group_ids": [(6, 0, cls.env.ref("base.group_user").ids)],
+                "field_readonly": True,
+                "condition_domain": "[('id', '>', 0)]",
+            }
+        )
+        cls.company_street_required = cls.env["custom.field.restriction"].create(
+            {
+                "required_model_id": cls.company_model_id.id,
+                "field_id": cls.company_street_field.id,
+                "group_ids": [(6, 0, cls.env.ref("base.group_user").ids)],
+                "required": True,
+                "condition_domain": "[('id', '>', 0)]",
+            }
+        )
+        cls.company_city_invisible = cls.env["custom.field.restriction"].create(
+            {
+                "invisible_model_id": cls.company_model_id.id,
+                "field_id": cls.company_city_field.id,
+                "group_ids": [(6, 0, cls.env.ref("base.group_user").ids)],
+                "field_invisible": True,
+                "condition_domain": "[('id', '>', 0)]",
+            }
+        )
+        cls.company_tree_view = cls.env["ir.ui.view"].create(
+            {
+                "name": "Test Tree",
+                "model": "res.company",
+                "type": "tree",
+                "mode": "primary",
+                "arch": "<tree><field name='city'/>"
+                "<field name='street'/>"
+                "<field name='name'/></tree>",
+            }
+        )
+        cls.view_users_form.arch = cls.view_users_form.arch.replace(
+            "</notebook>",
+            "</notebook><field name='company_ids'><tree><field name='city'/>"
+            "<field name='street'/>"
+            "<field name='name'/></tree></field>",
+        )
+
+    def test_model_form(self):
+        """onchange should set correct model"""
+
+        partner_form_view = self.view_model.default_view(
+            self.partner_model_id._name, "form"
+        )
+        with Form(self.partner_model_id, view=partner_form_view) as partner_form:
+            with partner_form.custom_required_restriction_ids.new() as new_line:
+                self.assertIsNotNone(new_line.required_model_id)
+                new_line.field_id = self.partner_name_field_id
+            with partner_form.custom_readonly_restriction_ids.new() as new_line:
+                self.assertIsNotNone(new_line.readonly_model_id)
+                new_line.field_id = self.partner_name_field_id
+            with partner_form.custom_invisible_restriction_ids.new() as new_line:
+                self.assertIsNotNone(new_line.invisible_model_id)
```

Comment:
**todo**: I think these should be `self.assertTrue`, no? Because a many2one will never return `None` even if it is empty.